src/server/hotspot_partition_calculator.cpp

```diff
@@ -180,44 +177,39 @@ void hotspot_partition_calculator::detect_hotkey_in_hotpartition(int data_type)
     const dsn::replication::detect_action::type action)
 {
     FAIL_POINT_INJECT_F("send_detect_hotkey_request", [](dsn::string_view) {});
-    auto request = dsn::make_unique<dsn::replication::detect_hotkey_request>();
-    request->type = hotkey_type;
-    request->action = action;
+
+    dsn::replication::detect_hotkey_request req;
+    req.type = hotkey_type;
+    req.action = action;
+    dsn::replication::detect_hotkey_response resp;
```

Comment:
Move closer to the place where you use it, just before `_shell_context->ddl_client->detect_hotkey `

Reply:
+1